### PR TITLE
Slip77 derivation

### PR DESCRIPTION
**Pull request by altafan**
This closes [#76](https://github.com/vulpemventures/liquid-js-monorepo/issues/76).
This closes [#77](https://github.com/vulpemventures/liquid-js-monorepo/issues/77).
This closes [#78](https://github.com/vulpemventures/liquid-js-monorepo/issues/78).

We should this squash with root after it is merged.

NOTE: the only one test case that I found is taken from libwally and refers to the one described in SLIP-21.
Since the derivation scheme of the master follows SLIP-77 you should not consider the expected values in SLIP-21 as the expected values of our unit test.

### CODE REVIEW

File: ts_src/index.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const seed = Buffer.alloc(32, 0);
+    const node = new Slip77(seed);
+    const masterKey = Buffer.concat([seed, key]);
+    node._masterKey = masterKey;
+    return node;
+  }
+
+  _masterKey: Buffer;
+
+  constructor(_seed: Buffer | string) {
```
Comment (tiero) [inline ts_src/index.ts]: I would use a fromSeed instead of using the constructor, more self decribing

File: ts_src/index.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  deriveBlindingPrivKey(_script: Buffer | string): Buffer {
+    typeforce(typeforce.anyOf('Buffer', 'String'), _script);
+    const script = Buffer.isBuffer(_script)
+      ? _script
+      : Buffer.from(_script, 'hex');
+    return hmacSHA256(this._masterKey.slice(32), [script]);
+  }
+
+  deriveBlindingPubKey(_script: Buffer | string): Buffer {
```
Comment (tiero) [inline ts_src/index.ts]: What's script?

File: ts_src/index.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+export interface Slip77Interface {
+  _masterKey: Buffer;
+  masterBlindingKey(): Buffer;
+  deriveBlindingPrivKey(script: Buffer | string): Buffer;
+  deriveBlindingPubKey(script: Buffer | string): Buffer;
+}
+
+export class Slip77 implements Slip77Interface {
+  static fromMasterBlindingKey(_key: Buffer | string): Slip77 {
```
Comment (tiero) [inline ts_src/index.ts]: how it is supposed to be passed the key? base58? etc..

File: ts_src/index.ts (hunk 2 above)
Comment (tiero) [reply ts_src/index.ts]: It is the "path" ?

Comment (altafan) [reply ts_src/index.ts]: it's the output `scriptPubkey`

File: ts_src/index.ts (hunk 3 above)
Comment (altafan) [reply ts_src/index.ts]: hex string or buffer.. no fancy encoding

**Pull request closed without merging**